libvast/src/system/read_query.cpp

```diff
@@ -98,13 +103,24 @@ read_query(const invocation& inv, std::string_view file_option,
                     read_query(*fname), *fname));
     if (*fname == "-")
       return read_query(std::cin);
+    if (has_query_stdin)
+      return caf::make_error(
+        ec::invalid_argument,
+        fmt::format("got query '{}' on the command line and '{}' via stdin",
```

Comment:
I think the error should be `"got query '{}' on the command line and '{}' as input filename"`, and not try to read from `std::cin`.